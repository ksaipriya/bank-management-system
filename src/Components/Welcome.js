import React from 'react'
import { useHistory } from 'react-router';
import { useAuth0 } from '@auth0/auth0-react'
import { Form, Button, Col, Container } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import './styles/welcome.css'
const Welcome = () => {
    let history = useHistory();
    const { logout, isAuthenticated } = useAuth0()
    
    return (
        <div className="container">
            {isAuthenticated &&
                <Container>
                    <Form.Row>
                        <Form.Group as={Col} className="logout">
                            <Button onClick={() => logout()}>Logout</Button>
                        </Form.Group>
                    </Form.Row>
                    <h4>Welcome to <span>XXXX</span> Bank</h4>
                    <Form>
                        <Form.Row>
                            <Form.Group className="register" as={Col}>
                                <Form.Label>New User?</Form.Label><br />
                                <Button variant="success" onClick={() => history.push('/register')}>Register</Button>
                            </Form.Group>
                            <Form.Group className="login" as={Col}>
                                <Form.Label>Already Registered?</Form.Label><br />
                                <Button variant="primary" onClick={() => history.push('/directlogin')}>Login</Button>
                            </Form.Group>
                        </Form.Row>
                    </Form>
                </Container>
            }
        </div>
    )
}
export default Welcome;
