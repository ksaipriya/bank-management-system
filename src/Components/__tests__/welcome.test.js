import React from 'react';
import { render } from '@testing-library/react';
import Welcome from '../Welcome.js'
import ShallowRenderer from 'react-test-renderer/shallow';
import Adapter from 'enzyme-adapter-react-16';
import { shallow ,configure} from 'enzyme';

configure({ adapter: new Adapter() });
describe('Test case for Success Component', () => {

    it('matches snapshot', () => {
        const shallowRenderer = new ShallowRenderer();
        shallowRenderer.render(<Welcome />);
        const result = shallowRenderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });

    it('check Logout button', () => {
        const mockCallBack = jest.fn();
        const button = shallow(<button onClick={mockCallBack}>Logout</button>);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    })

    it('check Register button', () => {
        const mockCallBack = jest.fn();
        const button = shallow(<button onClick={mockCallBack}>Register</button>);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    })

    it('check Login button', () => {
        const mockCallBack = jest.fn();
        const button = shallow(<button onClick={mockCallBack}>Login</button>);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    })

})