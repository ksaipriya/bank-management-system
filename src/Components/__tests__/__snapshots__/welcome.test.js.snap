// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for Success Component matches snapshot 1`] = `
<div
  className="container"
/>
`;
